api/proto/teleport/legacy/client/proto/authservice.proto

```diff
@@ -2190,6 +2190,35 @@ message UpsertTokenV2Request {
   }
 }
 
+// ProcessSAMLIdPRequestRequest is a request to create and sign the SAML IdP response
+// to a SAML IdP auth request.
+message ProcessSAMLIdPRequestRequest {
+  // Destination is the destination of the response.
+  string Destination = 1 [(gogoproto.jsontag) = "destination"];
+  // RequestID is the request ID.
+  string RequestID = 2 [(gogoproto.jsontag) = "request_id"];
+  // RequestTime is the time the request was made.
+  google.protobuf.Timestamp RequestTime = 3 [
+    (gogoproto.stdtime) = true,
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag) = "request_time,omitempty"
+  ];
+  // MetadataURL is the metadata URL of the SAML IdP.
+  string MetadataURL = 4 [(gogoproto.jsontag) = "metadata_url"];
+  // SignatureMethod is the signature method to use.
+  string SignatureMethod = 5 [(gogoproto.jsontag) = "signature_method"];
```

Comment:
is it necessary for this to be decided by the proxy? I'm assuming it may be picked by the library or something, but maybe better to let the auth pick if possible to keep this flexible based on the key type we have

Reply:
It's defined at the IdP level, and the signature method is displayed to service providers via the SAML IdP metadata. Basically, whatever signature method that the IdP uses should be supportable here. Basically, we have a need for the library and the backend to support this being configurable and it will be displayed via the metadata endpoint.

Long story short, yes, it does. Let me know if you want to discuss more over slack.